qiita_db/reference.py

```diff
@@ -113,54 +115,62 @@ def exists(cls, name, version):
         QiitaDBNotImplementedError
             If the method is not overwritten by a subclass
         """
-        conn_handler = SQLConnectionHandler()
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "reference_name=%s AND reference_version=%s)".format(cls._table),
-            (name, version))[0]
+        with TRN:
+            sql = """SELECT EXISTS(
+                        SELECT * FROM qiita.{0}
+                        WHERE reference_name=%s
+                            AND reference_version=%s)""".format(cls._table)
+            TRN.add(sql, [name, version])
+            return TRN.execute_fetchlast()
 
     @property
     def name(self):
-        conn_handler = SQLConnectionHandler()
-        return conn_handler.execute_fetchone(
-            "SELECT reference_name FROM qiita.{0} WHERE "
-            "reference_id = %s".format(self._table), (self._id,))[0]
-        _, basefp = get_mountpoint('reference')[0]
+        with TRN:
+            sql = """SELECT reference_name FROM qiita.{0}
```

Comment:
This is not converting from any id, is accessing to the reference_name attribute of the reference class.